app/api/search/specs/routes.spec.js

```diff
@@ -44,14 +44,10 @@ describe('search routes', () => {
       expect(response).toEqual(expectedResults);
     };
 
-    it('should have a validation schema', () => {
-      expect(routes.get.validation('/api/search')).toMatchSnapshot();
-    });
```

Comment:
It's good that we no longer test the validation structure.  But now we are not testing validation at all in this scenario.  How should we approach this?

Reply:
@RafaPolit that is a good question, do you have any suggestion ?

1.  test validations that belong to the route as an integration test.
2.  test the validations itself separately, this makes more sense when reusing validations obviously, i think this particular one is duplicated now on the export csv.

Reply:
@joan @RafaPolit 
Test the validation will be testing a configuration not an implementation.
Those are the kind of tests that only break when you are intentionally changing the configuration and not because by mistake you break the functionality, the concept is basically the same as doing a snapshot of the config. 

Integration tests between the route model and database are ok but still not great, the thing is that the method, the path, and the validation of the route are a contract between the client and the server, and the best way to have that tested is with the e2e.

Reply:
maybe open this on a new issue to implement separately as this was tested already wrong before this ?

e2e are hard to maintain, i dont see why this tests should be there when the react app has nothing to do with this apart from calling the endpoint.

data endpoint validation is the server side concern, i agree that testing the validation by comparing the object is the same bad idea as the snapshot testing, integration tests using supertest should be quick to develop and you dont really need to go till the DB, just test some failing cases, the db integration tests are already there.

Reply:
When we did the metadata object change we had validations and tests for them on the server-side of the app, but we had the app crashing in many places because the client was not properly updated, even if we add Supertest test, we can be in the same scenario.

Reply:
I guess there are 2 things to test, the positive, that is that the client - backend can communicate, and the negative that is that you cant inject data that is malformed.

Reply:
The full length of this discussion is to be discussed in #2934